common/sgx_workload/workload/ext_work_order_info_kme.cpp

```diff
@@ -232,3 +314,145 @@ bool ExtWorkOrderInfoKME::CheckAttestationSelf(
     // To be implemented
     return 0;
 }  // ExtWorkOrderInfoKME::CheckAttestationSelf
+
+void ExtWorkOrderInfoKME::CalculateWorkOrderKeyInfoHash(
+    WorkOrderKeyInfo wo_key_info, ByteArray& wo_key_info_hash) {
+
+    std::string concat_str = \
+        ByteArrayToBase64EncodedString(wo_key_info.encrypted_sym_key) +
+        ByteArrayToBase64EncodedString(wo_key_info.encrypted_wo_key) +
+        ByteArrayToBase64EncodedString(wo_key_info.encrypted_signing_key);
+
+    std::string hash1_str = ByteArrayToBase64EncodedString(
+        tcf::crypto::ComputeMessageHash(StrToByteArray(concat_str)));
+
+    // sort input-data-keys and output-data-keys based on indices
+    std::sort(wo_key_info.in_data_keys.begin(), wo_key_info.in_data_keys.end(),
+        [](tcf::WorkOrderData x, tcf::WorkOrderData y) {
+            return x.index < y.index;});
+
+    std::sort(wo_key_info.out_data_keys.begin(), wo_key_info.out_data_keys.end(),
+        [](tcf::WorkOrderData x, tcf::WorkOrderData y) {
+            return x.index < y.index;});
+
+    std::string in_data_hash_str;
+    for (auto d: wo_key_info.in_data_keys) {
+        in_data_hash_str += ByteArrayToBase64EncodedString(
+            tcf::crypto::ComputeMessageHash(d.decrypted_data));
+    }
+
+    std::string out_data_hash_str;
+    for (auto d: wo_key_info.out_data_keys) {
+        out_data_hash_str += ByteArrayToBase64EncodedString(
+            tcf::crypto::ComputeMessageHash(d.decrypted_data));
+    }
+    std::string final_hash_str = \
+        hash1_str + in_data_hash_str + out_data_hash_str;
+    wo_key_info_hash = tcf::crypto::ComputeMessageHash(
+        StrToByteArray(final_hash_str));
+}  // ExtWorkOrderInfoKME::CalculateWorkOrderKeyInfoHash
+
+ByteArray ExtWorkOrderInfoKME::CreateJsonWorkOrderKeys(
+    WorkOrderKeyInfo wo_key_info) {
```

Comment:
Add function comments. Add json structure in the comment.

Reply:
will add comment. JSON structure is available in header file.